content/03-reference/01-tools-and-interfaces/02-prisma-client/02-constructor.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: 'Constructor'
+metaTitle: 'Prisma Client Constructor (Reference)'
+metaDescription: 'This page describes the Prisma Client constructor parameters.'
+---
+
+## Overview
+
+This page describes the Prisma Client constructor parameters.
+
+## `datasources`
+
+Use the `datasources` constructor parameter to programatically override the URL of a datasource - for example, as part of an integration test. To override a datasource:
+
+1. Define a datasources in the `prisma.schema` file. In the following example, there is a single datasource named `my_database`:
+
+   ```prisma
+   datasource my_database {
+    provider = "sqlite"
+    url      = env("DATABASE_URL")
+   }
+   ```
+
+   The `DATABASE_URL` environment variable defines the following value:
+
+   ```
+   DATABASE_URL="file:./dev.db"
+   ```
+
+   > **Note**: You must re-generate the Prisma Client each time you add or rename a datasource. Datasource names are included in the generated client.
+
+2. Instantiate a client and override the connection string of the `my_database` datasource:
+
+   ```ts
+   import { PrismaClient } from '@prisma/client'
+
+   const prisma = new PrismaClient({
+     datasources: {
+   |       my_database: 'file:./dev_qa.db',
+     },
+   })
```

Comment:
The moment we document this, this is our public API.

There was an issue about this being inconsistent with the structure of the `datasources` block in the schema, but that was closed: https://github.com/prisma/prisma-client-js/issues/698

Are we sure we are ok with this mismatch and that it is not possible to override the `provider` here?

Ping @prisma/frontend

Reply:
We can’t add the provider here because the client is generated before and can't change dynamically here (need to generate a new client)

Reply:
As Joël mentioned, we can't have the provider here, that's why there is this "mismatch". So, in my opinion, we should keep it like this.
What we **could** do as long as there's only one datasource anyway: We could just have one arg called `connectionString`. However, that would not scale with the future of having multiple data sources.

Reply:
When I last looked at the structure it has, I could not connect the dots as it was missing the `url` thing. So I misunderstood `db` to be a real key instead of referring to the value from the schema.

I think
```
{
  datasources: {
    db: {
      url: 'file:dev.db'
    }
  }
}
```

would be better here.

Reply:
From my side, nothing speaks against that, it makes sense to me.
Nobody uses this API anyway yet, so we can just change it.
Any objections @Jolg42 ?

Reply:
Sorry @mhwelander for turning this into a product meeting :D

Reply:
I was also +1 to change it when I first encountered it :)

Reply:
Ok, let's change it and block @mhwelander until it is done 👍

Reply:
https://github.com/prisma/prisma/issues/2801

Reply:
I implemented it as @janpio suggested. So we can document it like this now 🙂

Reply:
@mhwelander Tim just unblocked this PR :)
It now need to be updated to reflect it like this (new url property inside db as an object)
```
{
  datasources: {
    db: {
      url: 'file:dev.db'
    }
  }
}
```

Reply:
Suggested the required change: https://github.com/prisma/docs/pull/480/files#r442789126